sky/spot/controller.py

```diff
@@ -166,7 +166,14 @@ def _check_signal(self) -> Optional[spot_utils.UserSignal]:
                     signal = spot_utils.UserSignal(signal)
                 # Remove the signal file, after reading the signal.
                 signal_file.unlink()
-        return signal
+        if signal is None:
+            return
+        if signal == spot_utils.UserSignal.CANCEL:
+            raise exceptions.SpotUserCancelledError(
+                f'User sent {signal.value} signal.')
+
+        raise exceptions.SpotUnknownSignalError(
```

Comment:
I'd vote for keeping exceptions more basic/leaner. Can we use a RuntimeError() for this case?

Reply:
Good point! Changed to RuntimeError.